OpenSim/Sandbox/HumphreysImplicitDynamics/forwardImplicitTest.cpp

```diff
@@ -0,0 +1,368 @@
+/* -------------------------------------------------------------------------- *
+ *                             OpenSim:  forwardImplicitTest.cpp                             *
+ * -------------------------------------------------------------------------- *
+ * The OpenSim API is a toolkit for musculoskeletal modeling and simulation.  *
+ * See http://opensim.stanford.edu and the NOTICE file for more information.  *
+ * OpenSim is developed at Stanford University and supported by the US        *
+ * National Institutes of Health (U54 GM072970, R24 HD065690) and by DARPA    *
+ * through the Warrior Web program.                                           *
+ *                                                                            *
+ * Copyright (c) 2005-2012 Stanford University and the Authors                *
+ *                                                                            *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may    *
+ * not use this file except in compliance with the License. You may obtain a  *
+ * copy of the License at http://www.apache.org/licenses/LICENSE-2.0.         *
+ *                                                                            *
+ * Unless required by applicable law or agreed to in writing, software        *
+ * distributed under the License is distributed on an "AS IS" BASIS,          *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.   *
+ * See the License for the specific language governing permissions and        *
+ * limitations under the License.                                             *
+ * -------------------------------------------------------------------------- */
+
+/* 
+ *  Below is an example of an OpenSim application that provides its own 
+ *  main() routine.  
+ */
+
+//==============================================================================
+//==============================================================================
+#include <OpenSim/OpenSim.h>
+
+using namespace OpenSim;
+using namespace SimTK;
+using namespace std;
+
+struct momentArmAndLength {
+    SimTK::Vec2 length;
+    SimTK::Vec2 momentArm;
+};
+
+
+
+
+struct bodyResidual {
+    double forceResidual = SimTK::NaN;
+    double velocityResidual = SimTK::NaN;
+    double forceMass = SimTK::NaN;
+    SimTK::Mat22 df_dy = {SimTK::NaN,SimTK::NaN,SimTK::NaN,SimTK::NaN};
+    SimTK::Mat22 df_dydot = {SimTK::NaN,SimTK::NaN,SimTK::NaN,SimTK::NaN};
+};
+
+
+
+
+// Get the Residual of the mass body.
+bodyResidual calcBodyImplicitResidual(Model& osimModel, State& s, double velocityGuess, double accelGuess, Vec2 muscleForces)
+{
+
+    // Tried to use new implicit interface:
+    //   (going to just use equation implementation for now)
+    //const double qDotGuess = 0.0; const double uDotGuess = 0.0;
+    // Set derivative guess.
+    //coord.setStateVariableDerivativeGuess(s, "value", qDotGuess);
```

Comment:
This doesn't belong in this WIP but the issue is that the signature is not what you're expecting; this method does not take the state. The first argument is the name of the state variable, and you're missing an argument for the full yDotGuess vector.